src/lib.rs

```diff
@@ -6,70 +6,204 @@
 //!
 //! ### Usage
-//! ```rust
+//! ```
 //! extern crate character_converter;
 //!
-//! use character_converter::CharacterConverter;
+//! use character_converter::*;
 //!
-//! let converter: CharacterConverter = CharacterConverter::new();
 //!
-//! let traditional_text = "欧洲";
-//! let simplified_text = "歐洲";
+//! let traditional_text = "歐洲";
+//! let simplified_text = "欧洲";
 //!
 //! // Check script
-//! let result_one: bool = converter.is_traditional(traditional_text);
-//! println!("{}", result_one); // --> true
+//! assert!(is_traditional(traditional_text));
 //!
-//! let result_two: bool = converter.is_simplified(traditional_text);
-//! println!("{}", result_two); // --> false
+//! assert!(!is_simplified(traditional_text));
 //!
 //! // Convert script
-//! let result_three: String = converter.traditional_to_simplified(traditional_text);
-//! println!("{}", result_three == simplified_text); // --> true
+//! let result_three = traditional_to_simplified(traditional_text);
+//! assert!(result_three == simplified_text);
 //!
-//! let result_four: String = converter.simplified_to_traditional(simplified_text);
-//! println!("{}", result_four == traditional_text); // --> true
+//! let result_four = simplified_to_traditional(simplified_text);
+//! assert!(result_four == traditional_text);
+//! ```
+#![cfg_attr(feature = "bench", feature(test))]
 
 extern crate bincode;
 
-mod character_converter;
-pub use self::character_converter::Converter as CharacterConverter;
+use std::collections::HashMap;
+
+use bincode::deserialize_from;
+use fst::raw::{Fst, Output};
+use once_cell::sync::Lazy;
+
+static T2S: Lazy<HashMap<String, String>> =
+    Lazy::new(|| deserialize_from(&include_bytes!("../data/t2s.profile")[..]).unwrap());
+static S2T: Lazy<HashMap<String, String>> =
+    Lazy::new(|| deserialize_from(&include_bytes!("../data/s2t.profile")[..]).unwrap());
+
+// create an fst containing all the keys
+static T2S_FST: Lazy<Fst<Vec<u8>>> = Lazy::new(|| {
+    let mut keys: Vec<_> = T2S.keys().collect();
+    keys.sort_unstable();
+    Fst::from_iter_set(keys).unwrap()
+});
+
+// create an fst containing all the keys
+static S2T_FST: Lazy<Fst<Vec<u8>>> = Lazy::new(|| {
+    let mut keys: Vec<_> = S2T.keys().collect();
+    keys.sort_unstable();
+    Fst::from_iter_set(keys).unwrap()
+});
+
+fn is_script(
+    raw: &str,
+    mapping: &HashMap<String, String>,
+    backup: &HashMap<String, String>,
+) -> bool {
+    for character in raw.chars() {
+        if mapping.get(&character.to_string()).is_none() {
+            if backup.get(&character.to_string()).is_some() {
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
+pub fn is_traditional(raw: &str) -> bool {
+    is_script(raw, &T2S, &S2T)
+}
+
+pub fn is_simplified(raw: &str) -> bool {
+    is_script(raw, &S2T, &T2S)
+}
+
+fn convert_script(raw: &str, mapping: &HashMap<String, String>, fst: &Fst<Vec<u8>>) -> String {
```

Comment:
BTW while we are breaking things, can't we change the function signature to return a `Cow<str>` and allocate only if needed?

Reply:
We can do this optimization in another pull request, for now, it can always return a `Cow::Owned` a.k.a. `String`.